### PR TITLE
Copy //dart-pkg/zircon|fuchsia from Topaz into the engine.

### PR DESCRIPTION
**Pull request by chinmaygarde**
Build rules and include paths have been updated as necessary. No other functional change.

### CODE REVIEW

File: shell/platform/fuchsia/dart-pkg/zircon/lib/src/system.dart [hunk 1]
```diff
@@ -0,0 +1,161 @@
...
+class System extends NativeFieldWrapperClass2 {
+  // No public constructor - this only has static methods.
+  System._();
+
+  // Channel operations.
+  static HandlePairResult channelCreate([int options = 0])
+      native 'System_ChannelCreate';
+  static HandleResult channelFromFile(String path)
+      native 'System_ChannelFromFile';
+  static int reboot()
```
Comment (zanderso) [inline shell/platform/fuchsia/dart-pkg/zircon/lib/src/system.dart]: Are you planning to include updates still happening in the Fuchsia tree like https://fuchsia-review.googlesource.com/c/topaz/+/275857 ?

File: shell/platform/fuchsia/dart-pkg/zircon/sdk_ext.gni [hunk 2]
```diff
@@ -0,0 +1,12 @@
+# Copyright 2013 The Flutter Authors. All rights reserved.
+# Use of this source code is governed by a BSD-style license that can be
+# found in the LICENSE file.
+
+zircon_sdk_ext_lib = "//topaz/public/dart-pkg/zircon/lib/zircon.dart"
+
+zircon_sdk_ext_files = [
+  zircon_sdk_ext_lib,
+  "//topaz/public/dart-pkg/zircon/lib/src/handle.dart",
```
Comment (zanderso) [inline shell/platform/fuchsia/dart-pkg/zircon/sdk_ext.gni]: are these right?

**Comment by zanderso** [comment]
Oh, also, are the years in the copyright headers correct?

File: shell/platform/fuchsia/dart-pkg/zircon/lib/src/system.dart (hunk 1 above)
Comment (chinmaygarde) [reply shell/platform/fuchsia/dart-pkg/zircon/lib/src/system.dart]: Yes. I am keeping a list of these patches in the "Patches to Backport" section of go/fdk-notes.

File: shell/platform/fuchsia/dart-pkg/zircon/sdk_ext.gni (hunk 2 above)
Comment (chinmaygarde) [reply shell/platform/fuchsia/dart-pkg/zircon/sdk_ext.gni]: I dont think this file is being referenced. I'll get rid of it.

**Comment by chinmaygarde** [comment]
> Oh, also, are the years in the copyright headers correct?

Yes. Verified with @Hixie

**Pull request merged**